src/python/pants/backend/codegen/tasks/antlr_gen.py

```diff
@@ -9,32 +9,45 @@
 import os
 import re
 
-from twitter.common.collections import OrderedSet
-
 from pants.backend.codegen.targets.java_antlr_library import JavaAntlrLibrary
-from pants.backend.codegen.tasks.code_gen import CodeGen
+from pants.backend.codegen.tasks.simple_codegen_task import SimpleCodegenTask
 from pants.backend.jvm.targets.java_library import JavaLibrary
 from pants.backend.jvm.tasks.nailgun_task import NailgunTask
-from pants.base.address import SyntheticAddress
-from pants.base.address_lookup_error import AddressLookupError
-from pants.base.build_environment import get_buildroot
 from pants.base.exceptions import TaskError
-from pants.util.dirutil import safe_mkdir
+from pants.java import util
+from pants.option.options import Options
 
 
 logger = logging.getLogger(__name__)
+_ANTLR_VERSIONS = {'antlr3': '//:antlr-3.4', 'antlr4': '//:antlr-4'}
 
-
-class AntlrGen(CodeGen, NailgunTask):
+class AntlrGen(SimpleCodegenTask, NailgunTask):
 
   class AmbiguousPackageError(TaskError):
     """Raised when a java package cannot be unambiguously determined for a JavaAntlrLibrary."""
 
+  class AntlrIsolatedCodegenStrategy(SimpleCodegenTask.IsolatedCodegenStrategy):
+    def find_sources(self, target):
+      sources = super(AntlrGen.AntlrIsolatedCodegenStrategy, self).find_sources(target)
+      return [source for source in sources if source.endswith('.java')]
+
+    def _do_execute_codegen(self, targets):
```

Comment:
refactored this out by changing the test to check for a task error with a regular expression matching the error text.